src/iterators.jl

```diff
@@ -51,17 +51,26 @@ struct CellCache{X,G<:AbstractGrid,DH<:Union{AbstractDofHandler,Nothing}}
 end
 
 function CellCache(grid::Grid{dim,C,T}, flags::UpdateFlags=UpdateFlags()) where {dim,C,T}
-    N = nnodes_per_cell(grid)
+    N = 0
+    if isconcretetype(getcelltype(grid))
+        N = nnodes_per_cell(grid)
+    end
     nodes = zeros(Int, N)
     coords = zeros(Vec{dim,T}, N)
     return CellCache(flags, grid, ScalarWrapper(-1), nodes, coords, nothing, Int[])
 end
 
 function CellCache(dh::DofHandler{dim}, flags::UpdateFlags=UpdateFlags()) where {dim}
-    N = nnodes_per_cell(get_grid(dh))
+    N = 0
+    if isconcretetype(getcelltype(get_grid(dh)))
+        N = nnodes_per_cell(get_grid(dh))
+    end
```

Comment:
And this will be resized anyway, but this way there would be no performance/allocation change for users of concrete grids. 
```suggestion
    N = nnodes_per_cell(get_grid(dh), 1)
```